sky/clouds/service_catalog/azure_catalog.py

```diff
@@ -41,6 +42,20 @@ def get_hourly_cost(instance_type: str,
     return common.get_hourly_cost_impl(_df, instance_type, region, use_spot)
 
 
+def get_vcpus_from_instance_type(instance_type: str) -> float:
+    df = _df[_df['InstanceType'] == instance_type]
+    cpu_info = df['capabilities'].str.replace('\'', '"').apply(json.loads)
+
+    def get_vcpus(x: List[Dict[str, str]]) -> str:
+        for attr in x:
+            if attr['name'] == 'vCPUs':
+                return attr['value']
+
+    vcpus = cpu_info.apply(get_vcpus)
+    assert len(set(vcpus)) == 1, df
```

Comment:
Instead of just `df`, a more descriptive assert failure message before printing the df would be nice :)

Reply:
Added.